src/SelfService/Infrastructure/Api/JsonSchema/SelfServiceJsonSchemaController.cs

```diff
@@ -121,6 +120,44 @@ public async Task<IActionResult> AddSchema(string id, [FromBody] AddSelfServiceJ
                 new ProblemDetails { Title = "Uncaught Exception", Detail = $"AddSchema: {e.Message}." }
             );
         }
-#pragma warning restore CS0162 // Unreachable code detected
+    }
+
+    [HttpPost("validate")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized, "application/problem+json")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest, "application/problem+json")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError, "application/problem+json")]
+    public IActionResult IsSchemaValid([FromBody] ValidateSelfServiceJsonSchemaRequest request)
+    {
+        if (request.Schema == null)
+            return BadRequest(new ProblemDetails { Title = "Invalid Schema", Detail = "Schema in request is null" });
+
+        try
+        {
+            _selfServiceJsonSchemaService.MustValidateJsonSchema(request.Schema.ToJsonString());
+        }
+        catch (InvalidJsonSchemaException e)
+        {
+            return BadRequest(
+                new ProblemDetails
+                {
+                    Title = "Invalid Schema",
+                    Detail = $"Schema in request is not a valid json schema: {e.Message}"
+                }
+            );
+        }
+        catch (Exception e)
+        {
+            return BadRequest(e);
+        }
+
+        return Ok();
+    }
+
+    [HttpGet("ids")]
```

Comment:
when is this route supposed to be used?

Reply:
I think I will just remove it. Just found it useful, but should be easy enough to add again when we actually need it.